File: Jonghyeok Kim/2022-06-17-Container-With-Most-Water.py
from typing import List

class Solution:
    def maxArea(self, height: List[int]) -> int:
        l_idx, r_idx = 0, len(height) - 1
        res = 0
        while l_idx < r_idx:
            bucket_height = min(height[l_idx],height[r_idx])
            area = (r_idx - l_idx) * bucket_height
            res = max(res, area)
            
            if height[l_idx] < height[r_idx]:
                l_idx += 1
            else:
                r_idx -= 1
        return res
    

File: Hyonsok Na/week2/2022-06-17.Container-With-Most-Water.java
class Solution {
    public int maxArea(int[] height) {
        int l = 0;
        int r = height.length-1;
        int max =0;
        int area =0;
        while(l<=r) {
            if(height[l] < height[r]) {
                area = height[l]*(r-l);
                if(area>max) max = area;
                l++;
            } else {
                area = height[r]*(r-l);
                if(area>max) max = area;
                r--;
            }
        }
        return max;
    }
}
